pkg/spec.go

```diff
@@ -35,3 +46,40 @@ type TargetItem struct {
 	StorageClass string `yaml:"storageClass"`
 	Region       string `yaml:"region"`
 }
+
+func RandomInt(min, max int) int {
+	return min + rand.Intn(max-min)
+}
+
+// Generate random string of given length
+func GenerateRandomString(length int) string {
+	rand.Seed(time.Now().UnixNano())
+	bytes := make([]byte, length)
+	for i := 0; i < length; i++ {
+		bytes[i] = byte(RandomInt(97, 122))
+	}
+	return string(bytes)
+}
+
+// Generate Specifications
+func GenerateSpec(NoOfTargetsPerSource int) Spec {
+	var spec Spec
+	spec.Source = Source{
+		Bucket:       "image-sync-" + GenerateRandomString(6),
+		Cos:          "cos-image-sync-test-" + GenerateRandomString(6),
+		Object:       "",
+		StorageClass: PlanSlice[RandomInt(0, len(PlanSlice))],
+		Region:       RegionSlice[RandomInt(0, len(RegionSlice))],
+	}
+
+	spec.Target = make([]TargetItem, 0)
+	for tgt := 0; tgt < NoOfTargetsPerSource; tgt++ {
+		spec.Target = append(spec.Target, TargetItem{
+			Bucket:       "image-sync-" + GenerateRandomString(6),
+			StorageClass: PlanSlice[RandomInt(0, len(PlanSlice))],
+			Region:       RegionSlice[RandomInt(0, len(RegionSlice))],
+		})
+	}
+
+	return spec
+}
```

Comment:
If this is used only for the test then we can move to `_test.go`

Reply:
These methods are used by both tests and sync command.

Reply:
I will change as per your suggestion

Reply:
This can be moved to some common place like pkg/utils if they are using in the different packages

Reply:
Methods moved to utils package